src/transformStateWithClones.js

```diff
@@ -6,8 +6,39 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    stateCopy = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties': {
+        stateCopy = Object.assign(stateCopy, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        stateCopy = {};
+
+        break;
+      }
+    }
+    arr.push(stateCopy);
```

Comment:
```suggestion
    arr.push({ ...stateCopy });
```
how to fix this coment. better to work with stateCope and push its clone
https://github.com/mate-academy/js_stateful-clones/pull/2706/files#r1309741194